chunks.go

```diff
@@ -298,44 +298,60 @@ type ChunkReader interface {
 // of series data.
 type chunkReader struct {
 	// The underlying bytes holding the encoded series data.
-	bs [][]byte
+	bs []ByteSlice
 
 	// Closers for resources behind the byte slices.
 	cs []io.Closer
 
 	pool chunks.Pool
 }
 
-// newChunkReader returns a new chunkReader based on mmaped files found in dir.
-func newChunkReader(dir string, pool chunks.Pool) (*chunkReader, error) {
+func newChunkReader(bs []ByteSlice, cs []io.Closer, pool chunks.Pool) (*chunkReader, error) {
```

Comment:
Is the `newChunkReader` used anywhere beside the `NewDirChunkReader` func? If no, I would inline this in `NewDirChunkReader`. Otherwise that magic validation part could be confusing (:

Reply:
Ops, no, `NewChunkReader` should actually call it too.